BAU: add pull request template (#1)
communitiesuk/funding-service-design-TEMPLATE: .github/PULL_REQUEST_TEMPLATE.md

_Add ticket reference to Pull Request title: e.g. 'FS-123: Add content', if there is no ticket prefix with BAU:_

## WHAT YOU DID
_A brief description of the pull request:_ 

> 

Reviewer: Should we have a discussion points section? IE things the merger(?) wants to alert the team about and document for future generations in the PR.

> ### JIRA link (if applicable)

Author: Can we remove this heading and just keep the text below?

Reviewer 2: Agree - changes made 👍

### Change description
_A brief description of the pull request_ 

- [ ] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)

> - [ ] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items to prevent them from overlapping the logo")

Reviewer 3: Sorry to be pedantic, but could we maybe slightly change this example?

It's 74 characters long, but the maximum amount of characters in a title is 72 - so it might be truncated during some git commands.

Reviewer 3: ```suggestion
- [ ] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")
```

_Add ticket reference to Pull Request title: e.g. 'FS-123: Add content', if there is no ticket prefix with BAU_

### Change description
_A brief description of the pull request_ 

- [ ] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)

> - [ ] JIRA ticket updated accordingly

Reviewer 4: does this just involve moving the ticket along the swimlanes, or is there anything else involved? (could update this line to state this more clearly?)

> - [ ] Commit messages are meaningful and follow good commit message guidelines

Reviewer 4: could an example of this be helpful?

### How to test
_If manual testing is needed, give suggested testing steps_

> ### How should it be reviewed? 

Reviewer 4: How would this section be used? should the change requester have a say on how their PR should be reviewed (conflict of interest and all that?)

Reviewer 2: @gidsg  your thoughts on Gio's point?
